repose-aggregator/components/filters/saml-policy-translation-filter/src/main/scala/org/openrepose/filters/samlpolicy/SamlPolicyTranslationFilter.scala

```diff
@@ -236,10 +261,46 @@ class SamlPolicyTranslationFilter @Inject()(configurationService: ConfigurationS
 
     if (Option(configuration).map(_.getPolicyAcquisition.getCache.getTtl) != Option(newConfiguration.getPolicyAcquisition.getCache.getTtl)) {
       cache = CacheBuilder.newBuilder()
-                          .expireAfterWrite(newConfiguration.getPolicyAcquisition.getCache.getTtl, TimeUnit.SECONDS)
-                          .build(new CacheLoader[String, XsltExecutable]() {
-                            override def load(key: String): XsltExecutable = getPolicy(key)
-                          })
+        .expireAfterWrite(newConfiguration.getPolicyAcquisition.getCache.getTtl, TimeUnit.SECONDS)
+        .build(new CacheLoader[String, XsltExecutable]() {
+          override def load(key: String): XsltExecutable = getPolicy(key)
+        })
+    }
+
+    try {
+      // Create a DOM XMLSignatureFactory that will be used to
+      // generate the enveloped signature.
+      fac = XMLSignatureFactory.getInstance("DOM")
+      // Create a Reference to the enveloped document (in this case,
+      // you are signing the whole document, so a URI of "" signifies
+      // that, and also specify the SHA1 digest algorithm and
+      // the ENVELOPED Transform.
+      val ref = fac.newReference(
+        "",
+        fac.newDigestMethod(DigestMethod.SHA1, null),
+        Collections.singletonList(fac.newTransform(Transform.ENVELOPED, null.asInstanceOf[TransformParameterSpec])),
+        null,
+        null)
+      // Create the SignedInfo.
+      si = fac.newSignedInfo(fac.newCanonicalizationMethod(
+        CanonicalizationMethod.INCLUSIVE,
+        null.asInstanceOf[C14NMethodParameterSpec]),
+        fac.newSignatureMethod(SignatureMethod.RSA_SHA1, null),
+        Collections.singletonList(ref))
+      // Load the KeyStore and get the signing key and certificate.
+      val creds = newConfiguration.getSignatureCredentials
+      val keyStoreFilename = FileUtilities.guardedAbsoluteFile(configRoot, creds.getKeystoreFilename).getAbsolutePath
+      logger.debug("Attempting to load keystore located at: {}", keyStoreFilename)
+      val ks = KeyStore.getInstance("JKS")
+      ks.load(new FileInputStream(keyStoreFilename), creds.getKeystorePassword.toCharArray)
+      keyEntry = ks.getEntry(creds.getKeyName, new KeyStore.PasswordProtection(creds.getKeyPassword.toCharArray)).asInstanceOf[KeyStore.PrivateKeyEntry]
+      val cert = keyEntry.getCertificate.asInstanceOf[X509Certificate]
+      // Create the KeyInfo containing the X509Data.
+      val kif = fac.getKeyInfoFactory
+      val xd = kif.newX509Data(List(cert.getSubjectX500Principal.getName, cert).asJava)
+      ki = kif.newKeyInfo(Collections.singletonList(xd))
+    } catch {
+      case e: Exception => throw new UpdateFailedException("Failed to load the signing credentials.", e)
```

Comment:
Is there not a tighter exception you can catch? This will hide NPEs and other code problems, where we really only want to catch config issues here.